Dockerfile

```diff
@@ -1,29 +1,25 @@
 FROM golang:1.16-buster AS build
 
-ADD go.mod /promql-cli/go.mod
-ADD go.sum /promql-cli/go.sum
+COPY go.mod /promql-cli/go.mod
+COPY go.sum /promql-cli/go.sum
 
 WORKDIR /promql-cli/
 RUN go mod download
 
-RUN apt-get update && apt-get install -y make
-
-ADD ./ /promql-cli/
+COPY ./ /promql-cli/
 ARG TARGETARCH
 RUN OS=linux ARCH=${TARGETARCH} INSTALL_PATH=/promql-cli/build/bin/ make install
 
-# TODO explore other base images here.
-# Requirements:
-# - small
-# - correct perms available for mounting and using a config file (config > cmdline flags), right now we just run as root... (see below)
-# I'm generally not a fan of alpine/busybox for a cmdline env but maybe minideb or similar?
-# Stock deb slim is pretty rad already :)
-FROM debian:buster-slim AS promql-cli
+FROM debian:bullseye-slim AS promql-cli
 COPY --from=build /promql-cli/build/bin/promql /bin/promql
 
 RUN apt-get update \
-  && apt-get install -y ca-certificates \
-  && rm -rf /var/lib/apt/lists/*
+  && apt-get install --no-install-recommends -y ca-certificates \
+  && rm -rf /var/lib/apt/lists/* /var/cache/*
```

Comment:
```bash
ive --ci promql-cli:dev
  Using default CI config
Image Source: docker://promql-cli:dev
Fetching image... (this can take a while for large images)
Analyzing image...
  efficiency: 99.4883 %
  wastedBytes: 722945 bytes (723 kB)
  userWastedPercent: 4.1128 %
Inefficient Files:
Count  Wasted Space  File Path
    2        322 kB  /var/log/lastlog
    2        168 kB  /var/lib/dpkg/status-old
    2        168 kB  /var/lib/dpkg/status
    2         35 kB  /var/log/faillog
    2         11 kB  /var/lib/apt/extended_states
    2        9.4 kB  /var/log/apt/eipp.log.xz
    2        1.9 kB  /etc/passwd
    2        1.8 kB  /etc/passwd-
    2        1.0 kB  /etc/shadow
    2        1.0 kB  /etc/shadow-
    2         907 B  /etc/group
    2         892 B  /etc/group-
    2         759 B  /etc/gshadow
    2          20 B  /etc/subuid
    2          20 B  /etc/subgid
    2           0 B  /tmp
    2           0 B  /var/lib/dpkg/triggers/Lock
    2           0 B  /var/lib/dpkg/triggers/Unincorp
    2           0 B  /var/lib/dpkg/lock-frontend
    2           0 B  /var/lib/dpkg/lock
    2           0 B  /var/lib/apt/lists
    2           0 B  /var/lib/dpkg/updates
    2           0 B  /var/cache/ldconfig
    2           0 B  /var/cache/apt
    2           0 B  /etc/.pwd.lock
Results:
  PASS: highestUserWastedPercent
  SKIP: highestWastedBytes: rule disabled
  PASS: lowestEfficiency
Result:PASS [Total:3] [Passed:2] [Failed:0] [Warn:0] [Skipped:1]

```